OneBusAway_WatchKit_Extension/OBAArrivalsAndDeparturesInterfaceController.m

```diff
@@ -0,0 +1,133 @@
+//
+//  OBAArrivalsAndDeparturesInterfaceController.m
+//  org.onebusaway.iphone
+//
+//  Created by Shannon Young on 4/14/15.
+//  Copyright (c) 2015 OneBusAway. All rights reserved.
+//
+
+#import "OBAArrivalsAndDeparturesInterfaceController.h"
+
+#import "OBAStopWK.h"
+#import "OBAArrivalAndDepartureWK.h"
+
+#import "OBAStopRowController.h"
+
+#define kStopRouteDetailRowType @"Route Detail Row"
+
+@interface OBAArrivalsAndDeparturesInterfaceController ()
+
+@property (nonatomic) NSMutableOrderedSet *arrivalsAndDepartures;
+
+@property (weak, nonatomic) NSURLSessionDataTask *dataTask;
+
+@end
+
+@implementation OBAArrivalsAndDeparturesInterfaceController
+
+- (void)awakeWithContext:(id)context {
+    [super awakeWithContext:context];
+
+    // Setup defaults
+    self.arrivalsAndDepartures = [NSMutableOrderedSet new];
+    self.maxCount = NSIntegerMax;
+    self.minutesBefore = 5;
+    self.minutesAfter = 35;
+}
+
+- (void)didDeactivate {
+    // This method is called when watch view controller is no longer visible
+    [super didDeactivate];
+
+    [self.dataTask cancel];
+    self.dataTask = nil;
+}
+
+- (NSArray *)filteredRouteIds {
+    return nil;
+}
+
+- (void)refreshArrivalsAndDeparturesForStop:(OBAStopWK *)stop {
```

Comment:
Is there any reason why this logic would be inside the controller and not the OBA "service layer"?

I'm somewhat sure that we probably already have this call in the main OBA service stack, no? Could be wrong.